### ISSUE TITLE
Feature request: add support for Durable functions in Decorators

### ISSUE THREAD

**Issue by ConnorKirk** [aws-powertools/powertools-lambda-typescript#4832]
### Expected Behavior

I am able to use Powertools class decorators with Durable Function handlers

### Current Behavior

Typescript emits a warning, as the Handler interface does not accept the DurableContext type.

The decorators break, as they try to access properties on the Lambda Context, but these must be accessed with a different path in the DurableContext type.

### Code snippet

```typescript
import {
  withDurableExecution,
  type DurableContext,
} from "@aws/durable-execution-sdk-js";
import { Logger, LogLevel } from "@aws-lambda-powertools/logger";

import { LambdaInterface } from "@aws-lambda-powertools/commons/types";

const logger = new Logger({
  serviceName: "test-suite-validation",
  logLevel: LogLevel.TRACE,
});

class Lambda implements LambdaInterface {
  // Decorate your handler class method
  @logger.injectLambdaContext()
  public async handler(
    _event: unknown,
    _context: DurableContext
  ): Promise<void> {
    logger.info("This is an INFO log with some context");
  }
}

const myFunction = new Lambda();
export const handler = withDurableExecution(
  myFunction.handler.bind(myFunction)
);
```

### Steps to Reproduce

1. Create a lambda function with a durable config specified, and the handler above
2. Invoke the function
3. Observe an error is raised

### Possible Solution

Update the Handler type in aws-lambda to accept a LambdaContext or a Durable Context

or

Create a new DurableHandler type?

Update decorators to correctly access the Lambda Context when passed a Durable Context

### Powertools for AWS Lambda (TypeScript) version

latest

### AWS Lambda function runtime

24.x

### Packaging format used

npm

### Execution logs

```Shell

```

**Comment by ConnorKirk** [comment]
I spoke with @svozza about this earlier today.
I see two parts to this issue.

1. We need to add logic to the decorators to detect if a context is a durable context. If so, we access the `lambdaContext` property before accessing the desired property. For example, [`injectLambdaContext`](https://github.com/aws-powertools/powertools-lambda-typescript/blob/12072217e316d7e4ffcd50ad53daae3d09b42742/packages/logger/src/Logger.ts#L505)
2. We need update the types affected by the new DurableContext.


We define an `AsyncHandler` type that extends the `@types/aws-lambda` `Handler` definition. 
* [commons - LambdaInterface](https://github.com/aws-powertools/powertools-lambda-typescript/blob/main/packages/commons/src/types/LambdaInterface.ts#L47)
* [definitelyTyped - aws-lambda - Handler](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/aws-lambda/handler.d.ts#L85)

We either need to update our extended definition, or get the `@types/aws-lambda` `Handler` definition updated to know about DurableContexts.

In either case, we need to decide whether to:
1. Add the durable execution SDK as a dev dependency, and use the original `DurableContext` type definition. This has the advantage of using the source of truth, at the cost of including a dependency for the sake of one type.
2. Define our own DurableContext type and keep it inline with the original.

**Comment by svozza** [comment]
I am strongly in favour of option 1: I'm happy to take the dev dependency as we won't be shipping it and it is the canonical version of the types associated with Durable Functions. Option 2 risks our definition becoming out of date and possibly causing runtime issues for customers due to changes that are not reflected in our custom type.

**Comment by dreamorosi** [comment]
I'm also in favor, although I'd like to propose we act as good citizens for the entire Node.js/JS community and push for the `@types/aws-lambda` in DefinitelyTyped to adopt both the DF types and the main Lambda types that I heard were published with Node.js 24 (unsure what's the package name).

This will be a more long term action, but it'll benefit everyone and not just Powertools customers.

**Comment by svozza** [comment]
Yes, fully agree that we should push this upstream to Definitely Typed.

**Comment by dreamorosi** [comment]
Despite the discussion about the types, which I still very much advocate for, I'm thinking that at least for now we just clearly document that we don't support the decorator pattern when working with Durable Functions.

If there's actual customer demand we can revisit the decision in the future.

**Comment by codyfrisch** [comment]
There is actually customer demand from me! I actually have a strong need to use durable functions, as it will drastically simplify significant parts of my code I want to refactor already.

Adjacent note, I personally think decorators should be isolated functions and not class methods. We just have to supply an instance to the function instead. This allows decorators to be tightly scoped (different one for durable functions etc.) and simple, without making the class code more difficult to manage.

**Comment by dreamorosi** [comment]
I agree that decorators should be isolated functions and that's something that we'd like to consider for our next major version this year.

I'll leave the issue open so that we consider implementing already as an isolated function.
